src/components/side-panel/layer-panel/layer-configurator.js

```diff
@@ -151,6 +153,42 @@ export default class LayerConfigurator extends Component {
             />
           </LayerConfigGroup>
         ) : null}
+
+        {/* text label */}
+        <LayerConfigGroup label={'text'}>
+          <PanelLabel>{`Show text label based on`}</PanelLabel>
+          <FieldSelector
+            fields={visConfiguratorProps.fields}
+            value={layer.config.textLabel.field && layer.config.textLabel.field.name || 'select a field'}
+            placeholder={'empty'}
+            onSelect={v => {
+              layerConfiguratorProps.onChange({
+                textLabel: {
+                  ...layer.config.textLabel,
+                  field: v
+                }
+              });
+            }}
+            erasable
+          />
+          <PanelLabel>{`Font size`}</PanelLabel>
```

Comment:
I'm not sure how to reuse VisConfigSlider since there's no visualChannel (?) for text label...(what's visual channel?)